tracks/python/exercises/armstrong-numbers/mentoring.md

```diff
@@ -0,0 +1,60 @@
+# Mentoring
+
+## Problem and challenges
+
+The problem requires the student to sum each digit of a number raised to the (length of number) power.
+
+## Reasonable solutions
+
+This exercise can be solved fairly straight forward:
+
+```python
+def is_armstrong(number):
+    digits = (int(d) for d in str(number))
+    count = len(str(number))
+    total = sum(digit ** count for digit in digits)
+    return number == total
+```
+
+This may also be written without the generator.
+This avoid the more complex generator type at the expense of a more complex the sum line.
+
+```python
+def is_armstrong(number):
+    count = len(str(number))
+    total = sum(int(digit) ** count for digit in str(number))
+    return number == total
+```
+
+### Common Suggestions
+
+#### Generator Expressions
+
+```python
+    total = sum([digit ** count for digit in digits])
+```
+
+Often students are unaware of [generator expressions](https://www.python.org/dev/peps/pep-0289/) and that they can be passed directly to functions that accept iterables.
+The code above uses a generator expression to build a list then iterates through the list.
+The middle step of a temporarily list is not needed.
+
+Potential feedback snippet:
+
+```
+`[f(a) for a in b]` iterates through `b` to create a list. You are then passing this list to a function which iterates through that list to build a result and discards that list. You can avoid that extra middle step of building and storing a list by passing a generator to the function instead and having the function process the elements of that iterator directly! You can do this either by doing `func(f(a) for a in b)` or `generator = (f(a) for a in b); func(generator)`
+```
```

Comment:
Consider avoiding "you" for "one", or "the code".

Such as "This passe this list to a function which iterates through that list to build a result and discard that list.  This avoids an extra middle step of building…"

It avoids unnecessary assignment of personality or ego and tends to focus more on the code for the reader, the voice being less _accusatory_.

Reply:
Changed.

Reply:
Was it changed somewhere else?